_posts/2018-07-27-Our-first-look-at-debos-new-Debian-images-generator.md

```diff
@@ -0,0 +1,285 @@
+things in a portable, easy-to-use container. In fact, this is one of the items
+from the
+[project's TODO list](https://github.com/go-debos/debos/blob/master/TODO#L12).
+Thanks to the `VirtualBox` triage I am already armed with a list of
+dependencies and potential issues to come.
+
+I've noticed that `fakemachine` already has a
+[Dockerfile](https://github.com/go-debos/fakemachine/blob/master/Dockerfile),
+so I had something to take a look at. However, it seems to be used to build
+`fakemachine` at container runtime, which is not exactly our target if we want
+to push the functional `debos` image to the `dockerhub`.
+
+I went with
+[multi-stage-buld](https://docs.docker.com/develop/develop-images/multistage-build/)
+in order to reduce image size (do not include build dependencies into the
+image, only the runtime ones).
+
+I ended up with
+[the following Dockerfile](https://github.com/3mdeb/debos/blob/add-dockerfile/docker/Dockerfile).
+
+```
+./docker/run.sh -t architecture:"amd64" doc/examples/example.yaml
+```
+
+```
+2018/07/26 18:36:39 Debootstrap (stage 2) | chroot: failed to run command '/debootstrap/debootstrap': Exec format error
+2018/07/26 18:36:39 debootstrap.log | gpgv: Signature made Thu Jul 26 14:21:51 2018 UTC
+2018/07/26 18:36:39 debootstrap.log | gpgv:                using RSA key A1BD8E9D78F7FE5C3E65D8AF8B48AD6246925553
+2018/07/26 18:36:39 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (7.0/wheezy) <ftpmaster@debian.org>"
+2018/07/26 18:36:39 debootstrap.log | gpgv: Signature made Thu Jul 26 14:21:51 2018 UTC
+2018/07/26 18:36:39 debootstrap.log | gpgv:                using RSA key 126C0D24BD8A2942CC7DF8AC7638D0442B90D010
+2018/07/26 18:36:39 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (8/jessie) <ftpmaster@debian.org>"
+2018/07/26 18:36:39 debootstrap.log | gpgv: Signature made Thu Jul 26 14:21:51 2018 UTC
+2018/07/26 18:36:39 debootstrap.log | gpgv:                using RSA key 16E90B3FDF65EDE3AA7F323C04EE7237B7D453EC
+2018/07/26 18:36:39 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>"
+2018/07/26 18:36:39 Action `debootstrap` failed at stage Run, error: exit status 126
+Powering off.
+```
+
+And the build for host architecture:
+
+```
+./docker/run.sh -t architecture:"amd64" doc/examples/example.yaml
+```
+
+Finished just fine:
+
+```
+2018/07/26 19:20:30 ==== overlay ====
+Overlaying /root/doc/examples/overlays/sudo on /scratch/root
+2018/07/26 19:20:30 ==== run ====
+2018/07/26 19:20:30 echo debian > /etc/hostname | host's /etc/localtime is not a symlink, not updating container timezone.
+2018/07/26 19:20:30 ==== pack ====
+2018/07/26 19:20:30 Compression to /root/debian-stretch-amd64.tgz
+Powering off.
+2018/07/26 19:20:44 ==== Recipe done ====
+```
+
+So, when building for non-host architecture, we have the same error as was in
```

Comment:
1.  Instead of `so` you can use:
- To summarize
- To sum up
- In a conclusion
- Long story short (rather unofficial for summarizing)
- In a nutshell

2. `... when building a non-host architecture...`

3. `... we have the same error as it was in...`